src/types/bytes.rs

```diff
@@ -88,6 +89,31 @@ impl PyBytes {
     /// Gets the Python string as a byte slice.
     #[inline]
     pub fn as_bytes(&self) -> &[u8] {
+        Py2Borrowed::from_gil_ref(self).as_bytes()
+    }
+}
+
+/// Implementation of functionality for [`PyBytes`].
+///
+/// These methods are defined for the `Py2<'py, PyBytes>` smart pointer, so to use method call
+/// syntax these methods are separated into a trait, because stable Rust does not yet support
+/// `arbitrary_self_types`.
+#[doc(alias = "PyBytes")]
+pub(crate) trait PyBytesMethods<'py> {
+    /// Gets the Python string as a byte slice.
+    fn as_bytes(&self) -> &[u8];
+}
+
+impl<'py> PyBytesMethods<'py> for Py2<'py, PyBytes> {
+    #[inline]
+    fn as_bytes(&self) -> &[u8] {
+        Py2Borrowed::from(self).as_bytes()
+    }
+}
+
+impl<'a> Py2Borrowed<'a, '_, PyBytes> {
+    /// Gets the Python string as a byte slice.
+    fn as_bytes(self) -> &'a [u8] {
```

Comment:
And Clippy is being a bit of a nag here. Maybe just `#[allow(clippy::wrong_self_convention)]`?